options/posix/include/mlibc/posix-sysdeps.hpp

```diff
@@ -155,15 +157,14 @@ int sys_vm_unmap(void *pointer, size_t size);
 [[gnu::weak]] int sys_accept(int fd, int *newfd, struct sockaddr *addr_ptr, socklen_t *addr_length);
 [[gnu::weak]] int sys_bind(int fd, const struct sockaddr *addr_ptr, socklen_t addr_length);
 [[gnu::weak]] int sys_connect(int fd, const struct sockaddr *addr_ptr, socklen_t addr_length);
-[[gnu::weak]] int sys_sockname(int fd, struct sockaddr *addr_ptr, socklen_t max_addr_length,
-		socklen_t *actual_length);
-[[gnu::weak]] int sys_peername(int fd, struct sockaddr *addr_ptr, socklen_t max_addr_length,
-	socklen_t *actual_length);
+[[gnu::weak]] int sys_sockname(int fd, struct sockaddr *addr_ptr, socklen_t *max_addr_length);
+[[gnu::weak]] int sys_peername(int fd, struct sockaddr *addr_ptr, socklen_t *max_addr_length);
```

Comment:
Why are these signatures changed?

Reply:
The previous approach looks much saner to me here.

Reply:
Well, IMHO the in-out argument (which uses the in-argument for a completely different purpose than the out-argument) is quite ugly. Just that Linux also does it that way doesn't mean that we should change the sysdeps to force every port to do it. IMHO Linux is "wrong" here.